lightning/src/ln/channel.rs

```diff
@@ -2969,6 +2969,27 @@ impl<Signer: Sign> Channel<Signer> {
 			panic!("Cannot update fee while peer is disconnected/we're awaiting a monitor update (ChannelManager should have caught this)");
 		}
 
+		// Before to propose a feerate update, check that we can actually afford the new fee compare to the holder balance.
+		let inbound_stats = self.get_inbound_pending_htlc_stats(Some(feerate_per_kw));
+		let outbound_stats = self.get_outbound_pending_htlc_stats(Some(feerate_per_kw));
+		let total_fee = feerate_per_kw as u64 * (COMMITMENT_TX_BASE_WEIGHT + (inbound_stats.pending_htlcs as u64 + outbound_stats.pending_htlcs as u64) * COMMITMENT_TX_WEIGHT_PER_HTLC) / 1000;
+		if self.value_to_self_msat / 1000 - total_fee - outbound_stats.pending_htlcs_value_msat < self.counterparty_selected_channel_reserve_satoshis.unwrap() {
```

Comment:
Fuzzer tells me this can underflow. This works but is ugly, either indent it or make temp variables:

```suggestion
		if (self.value_to_self_msat / 1000).checked_sub(total_fee).map(|a| a.checked_sub(outbound_stats.pending_htlcs_value_msat)).unwrap_or(None).unwrap_or(u64::max_value()) < self.counterparty_selected_channel_reserve_satoshis.unwrap() {
```

Reply:
`unwrap_or(u64::min_value)` here ? As we're subtracting, returning `u64::max_value` if `self < rhs` sounds wrong ?

Reply:
Yep, I believe you're right.